### PR TITLE
[AuthSamples] Update static files sample to use AuthZ middleware

### PR DESCRIPTION
**Pull request by JamesNK**
Update the static files sample to use AuthZ middleware

From https://github.com/aspnet/AuthSamples/pull/65

### CODE REVIEW

**Comment by JamesNK** [comment]
TODO: Address @Tratcher's feedback

**Comment by JamesNK** [comment]
🆙 📅 @Tratcher

File: src/AuthSamples/samples/StaticFilesAuth/Startup.cs [hunk 1]
```diff
@@ -114,81 +130,61 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, IAuthori
...
                 }
+            }
 
-                await next();
-            });
-            branch.UseFileServer(new FileServerOptions()
+            return null;
+        }
+
+        private class EndpointFeature : IEndpointFeature
```
Comment (Tratcher) [inline src/AuthSamples/samples/StaticFilesAuth/Startup.cs]: Wasn't this available in Http now?

**Review by Tratcher — CHANGES_REQUESTED** [review]
Net gain, -3 lines of code. It would be more interesting if this got integrated into StaticFiles.

File: src/AuthSamples/samples/StaticFilesAuth/Startup.cs (hunk 1 above)
Comment (JamesNK) [reply src/AuthSamples/samples/StaticFilesAuth/Startup.cs]: Extensions methods were added. Done.

I also optimized the logic some more and fixed the issue where it wouldn't work if a directory had no files in it.

**Review by Tratcher — APPROVED** [review]

**Pull request merged**